File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> counterPart = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; i++) {
            int currValue = nums[i];
            int difference = target - currValue;

            if (counterPart.containsKey(difference)) {
                return new int[] { counterPart.get(difference), i};
            }

            counterPart.put(currValue, i);
        }

        return new int[] {};
    }
}

File: Data Structures & Algorithms/sort-colors/submission-1.cs
public class Solution {
    public void SortColors(int[] nums) {
        int l = 0;
        int m = 0;

        for(int r = 0; r < nums.Length; r++) {
            int temp = nums[r];
            nums[r] = 2;

            if (temp < 2) {
                nums[m] = 1;
                m++;
            }
            if (temp < 1) {
                nums[l] = 0;
                l++;
            }
        }
    }
}